setup.py

```diff
@@ -17,10 +26,14 @@ def git_describe_version(original_version):
     _, gd_version = libver["git_describe_version"]()
     if gd_version is not None and gd_version != original_version:
         print("Use git describe based version %s" % gd_version)
+    if gd_version is None:
+        print(f"Use original version {original_version}")
+        return original_version
     return gd_version
 
 
-__version__ = git_describe_version(None)
+__version__ = get_lib_path()
+__version__ = git_describe_version(__version__)
```

Comment:
From this line, it looks like the `git_describe_version` would take the result of `get_lib_path()` and produce a description that combines both of them.  However, it instead falls back to the result of `get_lib_path()` whenever it would otherwise return `None`, and has no other usage.

Would the following be equivalent to the intended logic?  It might be more readable this way.

```python
def _get_version():
    git_version = git_describe_version()
    if git_version is not None:
        return git_version
        
    libinfo_version = get_lib_path()
    return libinfo_version
__version__ = get_version()
```

Reply:
I took this code from upstream mlc-llm and didn't change it to avoid merge conflict when we update our fork. Let me know if you think we should still apply these changes.

Reply:
In that case, good call, and best to avoid conflicts.